heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
@srkukarni can we merge our PR first, you merge with master, and continue with your refactor?